source/client/process_impl.cc

```diff
@@ -422,6 +434,12 @@ bool ProcessImpl::runInternal(OutputCollector& collector, const std::vector<UriP
     }
     int number_of_workers = determineConcurrency();
     shutdown_ = false;
+    envoy::config::bootstrap::v3::Bootstrap bootstrap;
+    createBootstrapConfiguration(bootstrap, uris, request_source_uri, number_of_workers);
+    // Needs to happen as early as possible (before createWorkers()) in the instantiation to preempt
+    // the objects that require stats.
+    store_root_.setTagProducer(Envoy::Config::Utility::createTagProducer(bootstrap));
```

Comment:
Does this need to happen conditionally, when the sink is enabled through config? Or will this be a true no-op otherwise?

Reply:
Add a condition to only call setTagProducer when there is sink enabled.

It is no-op (it sets the tag producer in the store to control tags) when there is no sink enabled. As in the Envoy example, it is called even when there is no sink enabled.
https://github.com/envoyproxy/envoy/blob/2a6a4b8d758b4bf4238cb6d2babdd3d83fbf45bc/source/server/server.cc#L324